### DISCUSSION TITLE
CLI Working - change end of carve behavior?

### DISCUSSION THREAD

**Discussion by Milner99** [winder/Universal-G-Code-Sender#1870] [Help]
Hi,

I've got the carve working from the command line - that's great, however at the end of the carve the CLI stays running in what I can assume is an idle state. 

Is it possible for the CLI to exit the app after the carve GCode instructions have been executed and the (for me) TinyG has reported all actions complete.

I'm looking to do this because I'm using a Python script to launch the CLI and would need to "detect" when the carve and all GCode is complete so I can instruct the machine to move into the next process (which is a different system).

Thank you for any help/guidance.

C

**Comment by breiler** [comment]
Just to make sure, what arguments are you using when running ugs-cli?

There is an argument `--daemon` or `-d` which will run UGS with a web server which would cause this. If you are not using that argument there is something else that is wrong.

**Reply by Milner99** [reply]
Thanks for replying @breiler 

I am not using -d or -daemon

I am using
--controller
--port
--baud
--print-stream
--file

<img width="1184" alt="Screenshot 2022-04-26 at 16 33 08" src="https://user-images.githubusercontent.com/1585406/165514833-56f41c50-7081-4125-8d46-23d7da4679a2.png">

I've rigged up a python script to monitor the realtime output of the UGS .jar and kill the process after the last line of the processed GCode which I am hoping will work the same way but the commandline call (in the screenshot) simply sits on a blank line at the end of the GCode instuctions - and even after all movement has been processed by the TinyG.

Happy to screen share if it will be useful for you?

Kind Regards

**Reply by breiler** [reply]
Nah, that is ok, I just wanted to make sure that you did not use that option.

This is not the intended behaviour, so this problem must have sneaked in in the last couple of releases. I don't have the time to look at it tonight though. Might take a couple of days before I can fix it.

**Reply by Milner99** [reply]
Thank you - I'll keep an eye on the repo and look for an update. Appreciate you taking the time to update.
